akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -218,7 +219,18 @@ object ShardRegion {
    */
   def gracefulShutdownInstance = GracefulShutdown
 
-  sealed trait ShardRegionQuery
+  /** Base marker trait for cluster sharding queries. */
+  private[akka] trait ShardingQuery
+  private[akka] trait ShardingQueryResponse
```

Comment:
```
* INTERNAL API
*/
@InternalApi private[akka] ...
```

Reply:
Won't `private[akka]` lead to warnings here because the subclasses are public? Could we make them sealed?

Reply:
Good point, then they should perhaps not be internal. I'm not 100 sure why we need them at all to be honest, the old one was to pattern match on a subset of the messages that the shard region accepts and have a separate receive-method for those.

Reply:
Yes thanks @jrudolph `sealed`. The `Get`* are public. 

@johanandren WRT matching the new `ShardingQuery` in this PR is for matching, similarly.

Reply:
Sorry, what I meant was that I'm not clear ont why it needs a name change, which is what makes it bin/source incompatible?

Reply:
I removed them and simplified. Though this wasn't a name change, it was the addition of a higher level trait for both cluster (`ShardingQuery`/`ShardingQueryResponse`) and region which I wanted earlier so the coordinator could match the cluster and regions match regions, when it came to that part of the message handling routes.